cdap-ui/app/features/dashboard/controllers/addwdgt-ctrl.js

```diff
@@ -50,17 +50,18 @@ function ($scope, $modalInstance, caskFocusManager, Widget) {
       'chart-11'
     ];
 
-    var widgets = [];
     if ($scope.model.metric.addAll) {
+      var widgets = [];
       // If the user chooses 'Add All' option, add all the metrics in the current context.
       angular.forEach($scope.model.metric.allMetrics, function(value) {
         widgets.push(
           new Widget({
             type: $scope.model.type,
-            title: value,
-            color: classes[(generateHash(value) * 13) % classes.length],
+            title: $scope.model.metric.title,
+            color: classes[(generateHash($scope.model.metric.context) * 13) % classes.length],
             metric: {
               context: $scope.model.metric.context,
+              names: [value],
```

Comment:
`names` is the list of metric names for the chart to poll.
`name` is the name of the widget (displays at the bottom).